draft-dekater-scion-controlplane.md

```diff
@@ -1578,106 +1582,162 @@ Many thanks go to William Boye (Swiss National Bank), Matthias Frei (SCION Assoc
+    // Core segment.
+    SEGMENT_TYPE_CORE = 3;
+  }
+
+
+service SegmentLookupService {
+    // Segments returns all segments that match the request.
+    rpc Segments(SegmentsRequest) returns (SegmentsResponse) {}
+}
+
+message SegmentsRequest {
+    // The source ISD-AS of the segment.
+    uint64 src_isd_as = 1;
+    // The destination ISD-AS of the segment.
+    uint64 dst_isd_as = 2;
+}
+
+message SegmentsResponse {
+    message Segments {
+        // List of path segments.
+        repeated PathSegment segments = 1;
+    }
+
+    // Mapping from path segment type to path segments. The key is the integer
+    // representation of the SegmentType enum.
+    map<int32, Segments> segments = 1;
+
+    // Deprecated list of signed revocations. Will be removed with header v1.
+    repeated bytes deprecated_signed_revocations = 1000;
+}
+
+service SegmentRegistrationService {
+    // SegmentsRegistration registers segments at the remote.
+    rpc SegmentsRegistration(SegmentsRegistrationRequest) returns (SegmentsRegistrationResponse) {}
+}
+
+message SegmentsRegistrationRequest {
+    message Segments {
+        // List of path segments.
+        repeated PathSegment segments = 1;
+    }
+
+    // Mapping from path segment type to path segments. The key is the integer
+    // representation of the SegmentType enum.
+    map<int32, Segments> segments = 1;
+}
+
+message SegmentsRegistrationResponse {}
+
+service SegmentCreationService {
+    // Beacon sends a beacon to the remote.
+    rpc Beacon(BeaconRequest) returns (BeaconResponse) {}
+}
+
+message BeaconRequest {
+    // Beacon in form of a partial path segment.
+    PathSegment segment = 1;
+}
+
+message BeaconResponse {}
+
+message PathSegment {
+    // The encoded SegmentInformation. It is used for signature input.
+    bytes segment_info = 1;
+    // Entries of ASes on the path.
+    repeated ASEntry as_entries = 2;
+}
+
+message SegmentInformation {
+    // Segment creation time set by the originating AS. Segment expiration time
+    // is computed relative to this timestamp. The timestamp is encoded as
+    // number of seconds elapsed since January 1, 1970 UTC.
+    int64 timestamp = 1;
+    // The 16-bit segment ID integer used for MAC computation.
+    uint32 segment_id = 2;
+}
+
+message ASEntry {
+    // The signed part of the AS entry. The body of the SignedMessage is the
+    // serialized ASEntrySignedBody. The signature input is defined as following:
+    //
+    //  input(ps, i) = signed.header_and_body || associated_data(ps, i)
+    //
+    //  associated_data(ps, i) = ps.segment_info ||
+    //                           ps.as_entries[1].signed.header_and_body ||
+    //                           ps.as_entries[1].signed.signature ||
+    //                           ...
+    //                           ps.as_entries[i-1].signed.header_and_body ||
+    //                           ps.as_entries[i-1].signed.signature
+    //
+    proto.crypto.v1.SignedMessage signed = 1;
```

Comment:
The `SignedMessage` message type and its internal details were previously included here, I think we need it. We can either add it back directly into the same code block, or we can create a separate block with just these helper types.
We need:
- SignedMessage 
- HeaderAndBodyInternal
- Header
- VerificationKeyID
- SignatureAlgorithm (previously missing).

Reply:
done